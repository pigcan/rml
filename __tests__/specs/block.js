
const MLTransformer = require('../../src/Transformer');

describe('block', () => {
  it('support simple', (done) => {
    new MLTransformer([
      `<div>`,
      `<block>`,
      `<div></div>`,
      `</block>`,
      `</div>`,
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support text', (done) => {
    new MLTransformer([
      `<block>`,
      `{{x}}`,
      `</block>`,
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });


  it('support nested block', (done) => {
    new MLTransformer(`
      <view r:if="{{z}}">
        <block>
          {{ z }}
        </block>
      </view>
`.trim())
      .transform((err, code) => {
        expect(code).toMatchSnapshot();
        done();
      });
  });

  it('support simple multiple', (done) => {
    new MLTransformer([
      `<div>`,
      `<block>`,
      `<div></div>`,
      `<div></div>`,
      `</block>`,
      `</div>`,
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support block if', (done) => {
    new MLTransformer([
      `<div>`,
      `<block r:if="{{a>b}}">`,
      `<div></div>`,
      `</block>`,
      `</div>`,
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support if nested', (done) => {
    new MLTransformer([
      `<div>`,
      `<block r:if="{{c>d}}">`,
      `<div r:if="{{a>b}}"></div>`,
      `</block>`,
      `</div>`,
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support block if multiple', (done) => {
    new MLTransformer([
      `<div>`,
      `<block r:if="{{a>b}}">`,
      `<div></div>`,
      `<div></div>`,
      `</block>`,
      `</div>`,
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support block for', (done) => {
    new MLTransformer([
      `<div>`,
      `<block r:for="{{a}}">`,
      `<div></div>`,
      `</block>`,
      `</div>`,
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support root block for', (done) => {
    new MLTransformer(`
<block r:for="{{z}}">
          z'\\
        </block>
`.trim())
      .transform((err, code) => {
        expect(code).toMatchSnapshot();
        done();
      });
  });
});
