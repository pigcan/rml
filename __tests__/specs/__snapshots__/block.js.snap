// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block allow empty 1`] = `
"import React from 'react';

export default function render(data) {
  return (
null
  );
};"
`;

exports[`block allow empty for 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    ((data.x) || []).map((item, index) => {
      return (
        null
      );
    })
  );
};"
`;

exports[`block allow empty if 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    (
    ((data.x)) ?
    (
    (
    ((data.x)) ?
    (
    null
    )
    :
    null
    )
    )
    :
    null
    )
  );
};"
`;

exports[`block allow empty in array 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    [
    <view>
    </view>
    ,
    null,
    ]
  );
};"
`;

exports[`block support block for 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      ((data.a) || []).map((item, index) => {
        return (
          <div>
          </div>
        );
      })
      }
    </div>
  );
};"
`;

exports[`block support block if 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      (
      ((data.a > data.b)) ?
      (
      <div>
      </div>
      )
      :
      null
      )
      }
    </div>
  );
};"
`;

exports[`block support block if multiple 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      (
      ((data.a > data.b)) ?
      (
      [
      <div>
      </div>
      ,
      <div>
      </div>
      ,
      ]
      )
      :
      null
      )
      }
    </div>
  );
};"
`;

exports[`block support if nested 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      (
      ((data.c > data.d)) ?
      (
      (
      ((data.a > data.b)) ?
      (
      <div>
      </div>
      )
      :
      null
      )
      )
      :
      null
      )
      }
    </div>
  );
};"
`;

exports[`block support nested block 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    (
    ((data.z)) ?
    (
    <view>
      {(data.z)}
    </view>
    )
    :
    null
    )
  );
};"
`;

exports[`block support root block for 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    ((data.z) || []).map((item, index) => {
      return (
        \`z'\\\\\\\\\`
      );
    })
  );
};"
`;

exports[`block support simple 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      <div>
      </div>
    </div>
  );
};"
`;

exports[`block support simple multiple 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      [
      <div>
      </div>
      ,
      <div>
      </div>
      ,
      ]
      }
    </div>
  );
};"
`;

exports[`block support text 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    (data.x)
  );
};"
`;
